src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+from typing import Dict, List, Tuple, Union
+
+from django.db import transaction
+
+from sentry.issues.escalating import query_groups_past_counts
+from sentry.issues.escalating_issues_alg import generate_issue_forecast
+from sentry.models import Group, GroupStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.tasks.base import instrumented_task
+from sentry.utils.json import JSONData
+
+BATCH_SIZE = 1000
+GroupForecastTuple = Tuple[Group, List[int]]
+ParsedGroupCount = Dict[int, Dict[str, List[Union[str, int]]]]
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.run_escalating_forecast",
+    queue="weekly_escalating_forecast",
+    max_retries=5,
+)  # type: ignore
+def run_escalating_forecast() -> None:
+    archived_groups = list(
+        Group.objects.filter(groupsnooze__until_escalating=True, status=GroupStatus.IGNORED)
+    )
+    if not archived_groups:
+        return
+
+    response = query_groups_past_counts(archived_groups)
+    group_counts = parse_groups_past_counts(response)
+    group_forecast_list = get_forecast_per_group(archived_groups, group_counts)
+
+    with transaction.atomic():
+        # Delete old forecasts
+        GroupForecast.objects.filter(group__in=archived_groups).delete()
```

Comment:
I chose to delete and then `bulk_create` since the `group` is the key and based on this [discussion](https://github.com/getsentry/sentry/pull/45971#discussion_r1144068135)
Since the `group` is the key, I think we'd need to change the `bulk_create` to make this work:

**Option 1:**
1. Check if the group exists in `GroupForecast` (which would still require `GroupForecast.objects.filter(group__in=archived_groups)`)
  a. If it does, add it to `bulk_update_list`
  b. Else add it to `bulk_create_list`
2. Bulk create and bulk update the lists

**Option 2:**
- Use `create_or_update` for each archived group

Do you think one of these would be more optimal?

Reply:
This is perfectly fine since it's within an atomic block.